docs/adrs/0078-api-versioning.md

```diff
@@ -0,0 +1,133 @@
+
+You can see an example of utilizing the generated code in the handlers and creating the necessary routing [in this commit](https://github.com/transcom/mymove/pull/10816/commits/0c4be3bab159a49f83e407830a9fecb9d27278da).
+
+#### Services Organization
+
+This is where there are a few options to organize our services. And I am currently unsure about the best way to go.
+
+Here are some questions to consider, when adding a new endpoint to our version 2 API:
+
+1. What do we do if we create a new endpoint that requires changes to the service that is called? How do we organize the functions/files? Do we create a new file in the services folder? Should we store these services files in a separate v2 subdir? What is the naming structure going to look like?
+2. What if the new endpoint does not require changes to the service? Should we make a new service anyway to prevent issues further down the line?
+
+
+* Option 1: Rename the old version to something that indicates it is old and add the new version to the same interface.
+  * Pros: As we remove the deprecated endpoints, we will have less code duplication.
+  * Cons: It will be easier to use the wrong service with the wrong endpoint.
+  * [Example can be seen here](https://github.com/transcom/mymove/pull/10816/commits/0913f2f4625450a2ac43b358e1890d17d77b747d)
+* Option 2: Create a new interface and service for the new version. This can be separated out into a new file or put in the same services file. In the example linked below it is separated out into its own file.
+  * Pros: Clear delineation between the two versions. You would not be able to mix up the new service with the old service since it does not use the same interface.
+  * Cons: Repetitive code. Naming questions arise, in order to delineate the new from the old. Difficulty when dealing with more complicated services, especially when we use a service in another service.
+  * [Example can be seen here](https://github.com/transcom/mymove/compare/0c4be3b...3769845)
+* Option 3: Create subdirectories in the current services directories for each version. Pass along a version flag in the AppContext (or some other method). Use that flag to identify which version of the service to use.
+  * Pros: Clear delineation between the two versions. There would not need to be a break from how we are currently utilizing our services and interfaces. We would only need to pull in the changed services into our subdirectories, which would lead to less code duplication.
+  * Cons: We would need to ensure that our flagging system worked completely as intended in order to not accidentally introduce breaking changes.
+  * [Example can be seen here](https://github.com/transcom/mymove/compare/0c4be3b...8ad4aed)
```

Comment:
**praise**: thank you for putting together this sample code! Really thank you for doing all the examples, but calling it out here since it was based an a suggestion late in the game. I think as you mention in the comments on the code, we'll need to figure out exactly how we want to work with the version in the context, but the example way you did it works to prove that this option can work.